extensions/wikia/PageHeader/Categories.class.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use CategoryHelper;
+use RequestContext;
+use Title;
+
+class Categories {
+	const VISIBLE_CATEGORIES_LIMIT = 4;
+
+	/**
+	 * @var Title[]
+	 */
+	public $categories;
+
+	/**
+	 * @var string
+	 */
+	public $inCategoriesText;
+
+	/**
+	 * @var string
+	 */
+	public $moreCategoriesText;
+
+	/**
+	 * @var string
+	 */
+	public $visibleCategoriesHTML;
+
+	public function __construct() {
+		$categories = $this->getCategories();
+
+		$this->categories = $categories;
+		$this->visibleCategoriesHTML = $this->getVisibleCategoriesHTML( $categories );
+		$this->moreCategories = $this->getMoreCategories( $categories );
+
+		$this->inCategoriesText = wfMessage( 'page-header-categories-in' )->escaped();
+		$this->moreCategoriesText = wfMessage( 'page-header-categories-more' )
+			->numParams( count( $this->moreCategories ) )
+			->escaped();
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function hasVisibleCategories() {
+		return count( $this->categories ) > 0;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function hasMoreCategories() {
+		return count( $this->categories ) > self::VISIBLE_CATEGORIES_LIMIT;
+	}
+
+	/**
+	 * @return Title[]
+	 */
+	private function getCategories() {
+		$categories = [];
+		$categoryNames = RequestContext::getMain()->getOutput()->getCategories();
+
+		foreach ( $categoryNames as $categoryName ) {
+			$categoryTitle = Title::newFromText( $categoryName, NS_CATEGORY );
+
+			if ( !$categoryTitle->isKnown() || CategoryHelper::getCategoryType( $categoryName ) === 'hidden' ) {
+				continue;
+			}
+
+			array_push( $categories, $categoryTitle );
+		}
+
+		return $categories;
+	}
+
+	/**
+	 * Prepare the HTML here instead of the template
+	 * so we can avoid having spaces before the commas
+	 * without using HTML comments or ugly formatting
+	 *
+	 * @param $categories Title[]
+	 *
+	 * @return string
+	 */
+	private function getVisibleCategoriesHTML( $categories ) {
+		$categoriesArray = array_slice( $categories, 0, self::VISIBLE_CATEGORIES_LIMIT - 1 );
+		$categoriesLinks = [];
+
+		/**
+		 * @var $category Title
+		 */
+		foreach ( $categoriesArray as $i => $category ) {
+			$categoriesLinks[] = '<a' .
```

Comment:
can we use some lib to build html tag instead of concatenating strings? eg. `Html::element('a', [...]);`